tests/unit/amo/components/TestLanguageTools.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+
+import LanguageTools, {
+  LanguageToolsBase,
+  LanguageToolList,
+} from 'amo/components/LanguageTools';
+import Link from 'amo/components/Link';
+import { ADDON_TYPE_DICT, ADDON_TYPE_LANG } from 'core/constants';
+import { loadAddonResults } from 'core/reducers/addons';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
+import {
+  createFakeLanguageAddon,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+import LoadingText from 'ui/components/LoadingText';
+
+
+describe(__filename, () => {
+  const addons = [
+    createFakeLanguageAddon({
+      name: 'Scottish Language Pack (with Irn-Bru)',
+      target_locale: 'en-GB',
+      type: ADDON_TYPE_LANG,
+    }),
+    createFakeLanguageAddon({
+      name: 'Old stuffy English',
+      target_locale: 'en-GB',
+      type: ADDON_TYPE_DICT,
+    }),
+    createFakeLanguageAddon({
+      name: 'English Language Pack with Extra Us',
+      target_locale: 'en-GB',
+      type: ADDON_TYPE_LANG,
+    }),
+    createFakeLanguageAddon({
+      name: 'Cool new English',
+      target_locale: 'en-US',
+      type: ADDON_TYPE_DICT,
+    }),
+    createFakeLanguageAddon({
+      name: 'le French Dictionary',
+      target_locale: 'fr',
+      type: ADDON_TYPE_DICT,
+    }),
+    createFakeLanguageAddon({
+      name: 'French Language Pack',
+      target_locale: 'fr',
+      type: ADDON_TYPE_LANG,
+    }),
+    createFakeLanguageAddon({
+      name: 'اُردو',
+      target_locale: 'ur',
+      type: ADDON_TYPE_DICT,
+    }),
+    createFakeLanguageAddon({
+      name: '正體中文 (繁體)',
+      target_locale: 'zh-TW',
+      type: ADDON_TYPE_LANG,
+    }),
+  ];
+
+  function renderShallow({
+    i18n = fakeI18n(),
+    store = dispatchClientMetadata().store,
+    ...props
+  } = {}) {
+    return shallowUntilTarget(
+      <LanguageTools i18n={i18n} store={store} {...props} />,
+      LanguageToolsBase
+    );
+  }
+
+  it('renders LoadingText if addons are not set', () => {
+    const root = renderShallow();
+
+    expect(root.find(LoadingText)).not.toHaveLength(0);
+  });
+
+  it('renders LoadingText if addons are empty', () => {
+    const { store } = dispatchClientMetadata();
+    store.dispatch(loadAddonResults({ addons: {} }));
+    const root = renderShallow({ store });
+
+    expect(root.find(LoadingText)).not.toHaveLength(0);
+  });
+
+  it('renders language tools in your locale', () => {
+    const { store } = dispatchClientMetadata({ lang: 'fr' });
+    store.dispatch(loadAddonResults({ addons }));
+    const root = renderShallow({ store });
+
+    const dictionary = root.find('.LanguageTools-in-your-locale-list-item')
+      .at(0);
```

Comment:
I don't think it's a good idea to rely on 0 and 1 because the order is not predictable (since it relies on `Object.values()`). Is there another way to separate dictionaries from lang packs?

Reply:
Good catch; I added a type classname to the elements to select them that way.